gwcs/wcs.py

```diff
@@ -794,6 +794,132 @@ def to_fits_sip(self, bounding_box=None, max_pix_error=0.25, degree=None,
+            A pair of tuples, each consisting of two numbers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+
+        bin_ext_name : str, optional
+            Extension name for the `~astropy.io.fits.BinTableHDU` extension.
+
+        crd_col_name : str, optional
+            Field name of the coordinate array in the structured array
+            stored in `~astropy.io.fits.BinTableHDU` data. This corresponds to
+            ``TTYPEi`` field in the FITS header of the binary table extension.
+
+        sampling : float, optional
+            The step, in input pixels, of the coordinate grid to be generated
+            for the coordinate array of the ``-TAB`` FITS WCS convention.
+
+        Returns
+        -------
+        hdr : astropy.io.fits.Header
+            Header with WCS-TAB information associated (to be used) with image
+            data.
+
+        bin_table : astropy.io.fits.BinTableHDU
+            Binary table extension containing the coordinate array.
+
+        Raises
+        ------
+        ValueError
+            When ``bounding_box`` is not defined either through the input
+            ``bounding_box`` parameter or this object's ``bounding_box``
+            property.
+
+        ValueError
+            When the number of ``sampling`` values is not 1 and also
+            does not match the number of image axes.
+
+        RuntimeError
+            If the number of image axes (`~gwcs.WCS.pixel_n_dim`) is larger
+            than the number of world axes (`~gwcs.WCS.world_n_dim`).
+
+        """
+        if bounding_box is None:
+            if self.bounding_box is None:
+                raise ValueError(
+                    "Need a valid bounding_box to compute the footprint."
+                )
+            bounding_box = self.bounding_box
+
+        if self.pixel_n_dim > self.world_n_dim:
+            raise RuntimeError(
+                "The case when the number of input axes is larger than the "
+                "number of output axes is not supported."
+            )
+
+        # 1D grid coordinates:
+        gcrds = []
+        cdel = []
+        sampling = np.atleast_1d(sampling).ravel()
+        if sampling.size != self.pixel_n_dim:
+            if sampling.size == 1:
+                sampling = np.repeat(sampling, self.pixel_n_dim)
+            else:
+                raise ValueError("Number of sampling values either must be 1 "
+                                 "or it must match the number of pixel axes.")
+
+        for (xmin, xmax), s in zip(bounding_box, sampling):
+            npix = max(2, 1 + int(np.ceil(abs((xmax - xmin) / s))))
+            gcrds.append(np.linspace(xmin, xmax, npix))
+            cdel.append((xmax - xmin) / (npix - 1))
+
+        # n-dim coordinate arrays:
+        coord = np.stack(
+            self(*np.meshgrid(*gcrds[::-1], indexing='ij')[::-1]),
+            axis=-1
+        )
+
+        # create primary header:
```

Comment:
It's not really a PrimaryHeader, right? If so, please change the comment to avoid confusion.

Reply:
Will fix this.